pkg/db/loader/issues.go

```diff
@@ -104,7 +107,15 @@ func findBugsForSearchStrings(isRegex bool, searchFor ...string) (map[string][]j
 		}
 		if err != nil {
 			lastUpdateError = err
-			log.Warnf("findBugsForSearchStrings got error in findBugs '%v' after %d requests", err, queryCtr)
+			findBugsErrorCount++
+			log.Warnf("findBugsForSearchStrings got error (%d of %d) in findBugs '%v' after %d requests",
+				findBugsErrorCount, maxFindBugErrors, err, queryCtr)
+			if findBugsErrorCount == maxFindBugErrors {
+				// If we exceed what we're willing to tolerate, finish doing search.ci lookups to avoid
+				// long delays for fetchdata uploads.
+				log.Errorf("findBugs calls with error exceeded max times; search.ci queries aborted")
+				break
```

Comment:
https://github.com/openshift/sippy/blob/376a0be3595f7f5ddda90e5f8834293ab93277e5/pkg/sippyserver/pr_commenting_processor.go#L275 has an example for rate limiting where the error count increases / decreases but it isn't really a generic for you could just drop in here.

Not saying you should change what you have just that sometime it can be good to track if the errors are successive or if there are some successful calls in between we can back off from the limit / failure case

Reply:
thanks.  Will keep that in mind for later.  In the scenario we have, the logs suggest about 4 hours of 30s retries.  I'll merge this and monitor.  The search.ci endpoint seems to be in good shape but I can still monitor for any of the new warnings.